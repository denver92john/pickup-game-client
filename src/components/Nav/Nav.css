.navbar {
    /*height: 50px;*/
    min-height: 40px;
    max-width: 650px;
    margin: auto;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.logo {
    width: 60px;
}

.nav-links {
    list-style-type: none;
}

.nav-links li {
    display: inline-block;
    margin-left: 5px;
}

.nav-link {
    color: var(--color6);
    font-size: var(--basis);
    padding: 5px;
}
.nav-link:active {
    background-color: var(--color6);
    color: var(--color1)
}
.nav-link:hover,
.nav-link:focus {
    border-bottom: 1px solid var(--color6);
    outline: none;
}

@media (min-width: 440px) {
    .navbar {
        max-width: 1230px;
    }
}