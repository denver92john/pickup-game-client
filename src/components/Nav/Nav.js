import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import TokenService from '../../services/token-service';
import logo from '../../images/PUG sports.png';
import './Nav.css';

export default class Nav extends Component {
    handleLogoutClick = () => {
        TokenService.clearAuthToken()
    }

    renderLoggedOutLinks() {
        return (
            <ul className="nav-links">
                <li>
                    <Link
                        to='/signup'
                    >
                        Signup
                    </Link>
                </li>
                <li>
                    <Link
                        to='/login'
                    >
                        Login
                    </Link>
                </li>
            </ul>
        );
    }

    renderLoggedInLinks() {
        return (
            <ul className="nav-links">
                <li>
                    <Link
                        to='/dashboard'
                    >
                        Dashboard
                    </Link>
                </li>
                <li>
                    <Link
                        to='/create'
                    >
                        Create
                    </Link>
                </li>
                <li>
                    <Link
                        to='/discover'
                    >
                        Discover
                    </Link>
                </li>
                <li>
                    <Link
                        onClick={this.handleLogoutClick}
                        to='/'
                    >
                        Logout
                    </Link>
                </li>
            </ul>
        );
    }

    render() {
        return <>
            <nav role="navigation" className="navbar">
                <div className="logo-div">
                    <Link
                        to='/'
                    >
                        <img src={logo} alt='pickup game logo' className='logo' />
                    </Link>
                </div>
                {TokenService.hasAuthToken()
                    ? this.renderLoggedInLinks()
                    : this.renderLoggedOutLinks()}
            </nav>
        </>
        
    }
}

/*function Nav() {
    return (
        <nav role="navigation" className="navbar">
            <div className="logo-div">
                <Link
                    to='/'
                >
                    <img src={logo} alt='pickup game logo' className='logo' />
                </Link>
            </div>

            <ul className="nav-links">
                <li>
                    <Link
                        to='/dashboard'
                    >
                        Dashboard
                    </Link>
                </li>
                <li>
                    <Link
                        to='/signup'
                    >
                        Signup
                    </Link>
                </li>
                <li>
                    <Link
                        to='/login'
                    >
                        Login
                    </Link>
                </li>
            </ul>
        </nav>
    );
}*/