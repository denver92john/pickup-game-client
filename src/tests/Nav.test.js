import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import Nav from '../components/Nav/Nav';

describe(`Nav component`, () => {
    it('renders without errors', () => {
        const nav = document.createElement('nav');
        ReactDOM.render(
            <BrowserRouter>
                <Nav />
            </BrowserRouter>,
            nav);
        ReactDOM.unmountComponentAtNode(nav);
    });

    it(`renders UI as expected`, () => {
        const tree = renderer
            .create(<BrowserRouter>
                <Nav />
            </BrowserRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})