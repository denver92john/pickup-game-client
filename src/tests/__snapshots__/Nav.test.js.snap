// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders UI as expected 1`] = `
<nav
  className="navbar"
  role="navigation"
>
  <div
    className="logo-div"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="pickup game logo"
        className="logo"
        src="PUG sports.png"
      />
    </a>
  </div>
  <ul
    className="nav-links"
  >
    <li>
      <a
        className="nav-link"
        href="/signup"
        onClick={[Function]}
      >
        Signup
      </a>
    </li>
    <li>
      <a
        className="nav-link"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </li>
  </ul>
</nav>
`;
